custom/icds_reports/reports/issnip_monthly_register.py

```diff
@@ -289,32 +368,23 @@ def agg_child_health_monthly(self):
             }, 'rations_21_plus_distributed')),
 
         )
-        return data
+        return {row['awc_id']: row for row in data}
 
     def get_awc_name(self, awc_id):
         return SQLLocation.objects.get(location_id=awc_id).name,
 
     @cached_property
     def to_pdf_format(self):
         for awc in self.config['awc_id']:
-            agg_awc_monthly_data = [x for x in self.agg_awc_monthly_data if x['awc_id'] == awc]
-            child_health_monthly_data = [x for x in self.child_health_monthly_data if x['awc_id'] == awc]
-            css_record_monthly = [x for x in self.css_record_monthly if x['awc_id'] == awc]
-            infrastructure_data = [x for x in self.infrastructure_data if x['awc_id'] == awc]
-            vhnd_data = [x for x in self.vhnd_data if x['awc_id'] == awc]
-            agg_child_health_monthly = [x for x in self.agg_child_health_monthly if x['awc_id'] == awc]
             yield dict(
                 awc_name=self.get_awc_name(awc),
-                agg_awc_monthly_data=agg_awc_monthly_data[0] if agg_awc_monthly_data else None,
-                child_health_monthly_data=child_health_monthly_data[0] if child_health_monthly_data else None,
-                css_record_monthly=css_record_monthly[0] if css_record_monthly else None,
-                infrastructure_data=infrastructure_data[-1] if infrastructure_data else None,
-                vhnd_data=vhnd_data[-1] if vhnd_data else None,
```

Comment:
I noticed that some of these were indexing `-1` while others were indexing `0` but it seems the code expects there to only ever be one of these anyways. was that just a legacy thing that was no longer actually doing anything?

Reply:
@czue, yes when we have ```-1``` we can have a situation where we can get more records but we want only newest